package/MDAnalysis/coordinates/GRO.py

```diff
@@ -269,19 +274,21 @@ def write(self, selection, frame=None):
            resName and atomName are truncated to a maximum of 5 characters
         """
         # write() method that complies with the Trajectory API
-        u = selection.universe
-        if frame is not None:
-            u.trajectory[frame]  # advance to frame
-        else:
-            frame = u.trajectory.ts.frame
 
-        # make sure to use atoms (Issue 46)
-        atoms = selection.atoms
-        # can write from selection == Universe (Issue 49)
-        coordinates = atoms.positions
+        try:
```

Comment:
AtomGroup might be missing `resnames` (for example) too.  Here you just need

```python
try:
    obj = obj.atoms
except AttributeError:
    if not isinstance(obj, Timestep):
        raise TypeError
```
Which converts the three possible inputs (U, AG, TS) into the two allowed (AG or TS)

Then going forward you need to treat AG and TS the same, which is possible with the `except (AE, NDE)` blocks